storage/database/dynamodb.go

```diff
@@ -347,7 +346,10 @@ func (dynamo *dynamoDB) Delete(key []byte) error {
 }
 
 func (dynamo *dynamoDB) Close() {
-	close(dynamo.quitCh)
+	dynamoOpenedDBNum--
+	if dynamoOpenedDBNum == 0 {
+		close(dynamoWriteCh)
+	}
```

Comment:
Is it necessary to wait stopping the loop?

Reply:
In other DB implementations (e.g. leveldb, badgerdb, memorydb), it is possible to close one db(e.g. header, state migratio). Also, state migration close one db when it is finishes.